providers/implementations/digests/sha2_prov.c

```diff
@@ -71,7 +71,12 @@ IMPLEMENT_digest_functions(sha224, SHA256_CTX,
 IMPLEMENT_digest_functions(sha256, SHA256_CTX,
                            SHA256_CBLOCK, SHA256_DIGEST_LENGTH, SHA2_FLAGS,
                            SHA256_Init, SHA256_Update, SHA256_Final)
-
+# ifndef FIPS_MODULE
```

Comment:
The space between `#` and `ifndef` should be removed. Sorry, for misleading you.